app/core/models/ocelot/stanzas/CalculationStanza.scala

```diff
@@ -125,16 +158,54 @@ sealed trait Operation {
 
 case class AddOperation(left: String, right: String, label: String) extends Operation {
 
-  def eval(labels: Labels): Labels = op(_ + _, (s1:String, s2:String) => Some(s1 + s2), unsupportedOperation("Add"), labels)
+  def eval(labels: Labels): Labels = {
 
+    val updatedLabels: Labels = operands(labels) match {
```

Comment:
updateLabels not needed, the rhs is the return value, also removes the need for the {} on eval()

Reply:
Done. Updatedlabels was probably a hangover from running the debugger on the tests.